<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InboundEditor</class>
 <widget class="QDialog" name="InboundEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>874</width>
    <height>497</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Inbound Editor</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="7,3">
   <item row="2" column="0">
    <layout class="QFormLayout" name="formLayout_4">
     <item row="0" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Tag</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="inboundTagTxt">
       <property name="placeholderText">
        <string>Tag of this inbound entry</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Listening</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0,1">
       <item>
        <widget class="QLineEdit" name="inboundHostTxt">
         <property name="placeholderText">
          <string>Hostname or IP Address</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="inboundPortTxt">
         <property name="placeholderText">
          <string>Port: 1080|env:PORT|80-85</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Protocol</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QComboBox" name="inboundProtocolTxt">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <item>
        <property name="text">
         <string>HTTP</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>SOCKS</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Dokodemo</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>MTProto</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Allocation Settings</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Strategy</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="strategyCombo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>Always</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Random</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Refresh</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="refreshNumberBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>2</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Concurrency</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="concurrencyNumberBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="1">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Sniffing Settings</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Destination
Override</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QListWidget" name="destOverrideList">
        <item>
         <property name="text">
          <string notr="true">http</string>
         </property>
         <property name="checkState">
          <enum>Unchecked</enum>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">tls</string>
         </property>
         <property name="checkState">
          <enum>Unchecked</enum>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="enableSniffingCB">
        <property name="text">
         <string>Enabled</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_29">
        <property name="text">
         <string>Enabled</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="HTTPPage">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>HTTP Inbound Settings</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,6">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Timeout</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="httpTimeoutSpinBox"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Allow Transparent</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="httpTransparentCB">
              <property name="text">
               <string>Enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>User Level</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="httpUserLevelSB"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="1" column="3">
             <widget class="QToolButton" name="httpRemoveUserBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../resources.qrc">
                <normaloff>:/icons/remove_connection_btn.png</normaloff>:/icons/remove_connection_btn.png</iconset>
              </property>
             </widget>
            </item>
            <item row="1" column="0" rowspan="2" colspan="3">
             <widget class="QListWidget" name="httpAccountListBox"/>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="httpAddPasswordTxt">
              <property name="placeholderText">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QToolButton" name="httpAddUserBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../resources.qrc">
                <normaloff>:/icons/add_connection_btn.png</normaloff>:/icons/add_connection_btn.png</iconset>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="httpAddUserTxt">
              <property name="placeholderText">
               <string>Username</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="5">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Accounts</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="SOCKSPage">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>SOCKS Inbound Settings</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="4,6">
          <item>
           <layout class="QFormLayout" name="formLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Auth</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="socksAuthCombo">
              <item>
               <property name="text">
                <string>No Auth</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Password</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Enable UDP</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>Local UDP IP</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_21">
              <property name="text">
               <string>User Level</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QCheckBox" name="socksUDPCB">
              <property name="text">
               <string>Enabled</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="socksUDPIPAddrTxt">
              <property name="placeholderText">
               <string>127.0.0.1</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="socksUserLevelSB"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="3">
             <widget class="QToolButton" name="socksRemoveUserBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../resources.qrc">
                <normaloff>:/icons/remove_connection_btn.png</normaloff>:/icons/remove_connection_btn.png</iconset>
              </property>
             </widget>
            </item>
            <item row="1" column="0" rowspan="2" colspan="3">
             <widget class="QListWidget" name="socksAccountListBox"/>
            </item>
            <item row="3" column="2">
             <widget class="QLineEdit" name="socksAddPasswordTxt">
              <property name="placeholderText">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QToolButton" name="socksAddUserBtn">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset resource="../../resources.qrc">
                <normaloff>:/icons/add_connection_btn.png</normaloff>:/icons/add_connection_btn.png</iconset>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="socksAddUserTxt">
              <property name="placeholderText">
               <string>Username</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="5">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Accounts</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="DokodemoPage">
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Dokodemo-Door Inbound Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>IP Address</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="dokoIPAddrTxt">
            <property name="placeholderText">
             <string>Not necessary when setting &quot;Follow Redirect&quot;</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Port</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="dokoPortSB">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>10086</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Network</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QCheckBox" name="dokoTCPCB">
              <property name="text">
               <string>TCP</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="dokoUDPCB">
              <property name="text">
               <string>UDP</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>Timeout</string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Follow Redirect</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>User Level</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="dokoTimeoutSB">
            <property name="minimum">
             <number>-1</number>
            </property>
            <property name="maximum">
             <number>8192</number>
            </property>
            <property name="value">
             <number>300</number>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QCheckBox" name="dokoFollowRedirectCB">
            <property name="text">
             <string>CheckBox</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QSpinBox" name="dokoUserLevelSB"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="MTProtoPage">
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>MTProto Inbound Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_7">
          <item row="0" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>EMail Address</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mtEMailTxt">
            <property name="placeholderText">
             <string notr="true">your.name@domain.com</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>Secret</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="mtSecretTxt">
            <property name="placeholderText">
             <string>SECRET</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>User Level</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="mtUserLevelSB"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>InboundEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>InboundEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
